libshaderc_util/include/libshaderc_util/spirv_tools_wrapper.h

```diff
@@ -34,6 +34,22 @@ bool SpirvToolsAssemble(const string_piece assembly, spv_binary* binary,
 // *text_or_error.
 bool SpirvToolsDisassemble(const std::vector<uint32_t>& binary,
                            std::string* text_or_error);
+
+// Used to tell SpirvToolsOptimize() whether to enable a certain pass.
+//
+// TODO(antiagainst): This may need to consider pass order once we have the
+// interface ready for users to specify optimization on a per pass base.
+// But now let's be simple.
+struct EnableOptimizationPass {
```

Comment:
Good suggestion! I've changed the impl. to use `std::vector` and now we have order support.

I've also added support for unify-constant, but have no only of testing it since this is optimizing the code compiled via glslang directly.